packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -195,12 +195,16 @@ class _GameWidgetState<T extends Game> extends State<GameWidget<T>> {
   late T currentGame;
 
   Future<void> get loaderFuture => _loaderFuture ??= (() async {
-        assert(currentGame.hasLayout);
-        final onLoad = currentGame.onLoadFuture;
+        final game = currentGame;
+        assert(game.hasLayout);
+        final onLoad = game.onLoadFuture;
         if (onLoad != null) {
           await onLoad;
         }
-        currentGame.onMount();
+        game.onMount();
+        if (game is FlameGame) {
+          game.setMounted();
+        }
```

Comment:
@st-pasha @renancaraujo is this the way to go, or do you have any other ideas?

Reply:
I wanted to suggest to set this flag directly in `onMount()`, but then I'm not even sure what `onMount` even means for the `Game` class (if it's distinct from attaching to the widget tree). So, perhaps we could set this flag at the end of the `onLoad()`?

Frankly, I'm completely confused here.

Reply:
`onLoad` should only run once of the game is removed from the widget and added again, so that would not render the correct behaviour. And running it directly in `onMount` would mean that the user would need to make sure they always run `super.onMount` last to get the correct behaviour, so I think the current solution is better than that.

Reply:
We can go with the current approach for now, and then if we think of a better way in the future, we can always refactor.

Reply:
I see that `FlameGame` now calls `setMounted()` inside its `mount()` method -- so perhaps these lines are no longer needed?

Reply:
Missed removing that one, removed now.

Reply:
@st-pasha everything should be updated now, can you have another look? :)